packages/astro/src/core/app/index.ts

```diff
@@ -175,14 +188,21 @@ export class App {
 		if (route.type === 'redirect') {
 			return RedirectSinglePageBuiltModule;
 		} else {
-			const importComponentInstance = this.#manifest.pageMap.get(route.component);
-			if (!importComponentInstance) {
-				throw new Error(
-					`Unexpectedly unable to find a component instance for route ${route.route}`
-				);
+			if (this.#manifest.pageMap) {
+				const importComponentInstance = this.#manifest.pageMap.get(route.component);
+				if (!importComponentInstance) {
+					throw new Error(
+						`Unexpectedly unable to find a component instance for route ${route.route}`
+					);
+				}
+				const pageModule = await importComponentInstance();
+				return pageModule;
+			} else if (this.#manifest.pageModule) {
+				const importComponentInstance = this.#manifest.pageModule;
+				return importComponentInstance;
+			} else {
+				throw new AstroError(AstroErrorData.FailedToFindPageMapSSR);
```

Comment:
I'm not sure if `AstroError` can be used in the `App`. IIUC this code is bundled into the user's server code. Checking it locally, I can see the entire AstroError handling is included so that might not be what we want as it increases the bundle size.

I think we can use a plain ol error here for now.